### ISSUE TITLE
Cannot convert undefined or null to object

### ISSUE THREAD

**Issue by tfrijsewijk** [whitespace-se/storybook-addon-html#46]
Hi, I installed your addon on Storybook 6.2.1. I'm running Node 12 and when opening Storybook I get the following error:

```
Uncaught TypeError: Cannot convert undefined or null to object
    at getPrototypeOf (<anonymous>)
    at r.parse (http://localhost:56006/vendors~main.manager.bundle.js:137897:23)
    at Object.t.parse (http://localhost:56006/vendors~main.manager.bundle.js:137921:50)
    at Mn (http://localhost:56006/vendors~main.manager.bundle.js:137963:22)
    at Un (http://localhost:56006/vendors~main.manager.bundle.js:138152:17)
    at Object.parse (http://localhost:56006/vendors~main.manager.bundle.js:138198:16)
    at Object.parse (http://localhost:56006/vendors~main.manager.bundle.js:138268:122695)
    at ts (http://localhost:56006/vendors~main.manager.bundle.js:138268:138882)
    at us (http://localhost:56006/vendors~main.manager.bundle.js:138268:141841)
    at http://localhost:56006/vendors~main.manager.bundle.js:138298:42079

The above error occurred in the <HTMLPanel> component:
    in HTMLPanel
    in div (created by AddonPanel)
    in AddonPanel
    in div (created by Context.Consumer)
    in Styled(div)
    in div (created by Context.Consumer)
    in Styled(div)
    in Unknown
    in Unknown
    in Unknown
    in Unknown (created by ManagerConsumer)
    in ManagerConsumer (created by Panel)
    in Panel (created by Layout)
    in div (created by Context.Consumer)
    in Styled(div) (created by Panel)
    in Panel (created by Layout)
    in div (created by Context.Consumer)
    in Styled(div) (created by Main)
    in div (created by Context.Consumer)
    in Styled(div) (created by Main)
    in Main (created by Layout)
    in Layout (created by Context.Consumer)
    in WithTheme(Layout)
    in Unknown
    in div (created by Context.Consumer)
    in Styled(div)
    in Unknown
    in Unknown (created by SizeMeRenderer(Component))
    in SizeMeReferenceWrapper (created by SizeMeRenderer(Component))
    in SizeMeRenderer(Component) (created by SizeMe(Component))
    in SizeMe(Component)
    in ThemeProvider
    in Unknown (created by ManagerConsumer)
    in ManagerConsumer (created by Manager)
    in EffectOnMount (created by Manager)
    in Manager (created by Context.Consumer)
    in Location (created by QueryLocation)
    in QueryLocation (created by Root)
    in LocationProvider (created by Root)
    in HelmetProvider (created by Root)
    in Root

React will try to recreate this component tree from scratch using the error boundary you provided, LocationProvider.
```

I also posted a message in a bug of the prettier repo: https://github.com/prettier/prettier/issues/10102#issuecomment-811921979

**Comment by thorn0** [comment]
Upgrade Babel. This was fixed in [v7.12.13 (2021-02-03)](https://github.com/babel/babel/blob/main/CHANGELOG.md#v71213-2021-02-03) by https://github.com/babel/babel/pull/12544

**Comment by ml-devninja** [comment]
Please notice that you need to target this specified version: `7.12.13`. As i had installed `7.13.14` of `@babel/core` this bug appears again.

**Comment by thorn0** [comment]
@ml-devninja You need to upgrade the other `@babel/*` packages too. The changelog says this was a change in the following packages: babel-helper-create-class-features-plugin, babel-helper-replace-supers, babel-plugin-transform-classes

**Comment by ml-devninja** [comment]
@thorn0 but actually this is how my `package.json` looks like:

```
  "devDependencies": {
    "@babel/core": "7.12.13",
    "@storybook/addon-actions": "^6.2.3",
    "@storybook/addon-essentials": "^6.2.3",
    "@storybook/addon-links": "^6.2.3",
    "@storybook/html": "^6.2.3",
    "@whitespace/storybook-addon-html": "^5.0.0",
    "babel-loader": "^8.2.2",
    "sass": "^1.32.4",
    "sass-loader": "^10.1.1"
  },
  "dependencies": {},
```
So for me only lock on specyfic `@babel/core` version helps

**Comment by tfrijsewijk** [comment]
Thanks for responsing!

Upgrading `@babel/core` wasn't enough for me as well. I'm using `@storybook/web-components` (instead of `@storybook/html`) and using Yarn Workspaces.

I fixed it by adding resolutions to the root package.json:

```
{
  [...]
  "resolutions": {
    "@babel/helper-create-class-features-plugin": "^7.12.13",
    "@babel/helper-replace-supers": "^7.12.13",
    "@babel/plugin-transform-classes": "^7.12.13"
  }
}
```

Just running `yarn` wasn't enough, I had to remove node_modules (`npx rimraf node_modules`) first (I'm using Yarn 2 without PnP).

I tried scoping the resolutions to `@whitespace/storybook-addon-html`:

```
{
  [...]
  "resolutions": {
    "@whitespace/storybook-addon-html/**/@babel/helper-create-class-features-plugin": "^7.12.13",
    "@whitespace/storybook-addon-html/**/@babel/helper-replace-supers": "^7.12.13",
    "@whitespace/storybook-addon-html/**/@babel/plugin-transform-classes": "^7.12.13"
  }
}
```

But Yarn 2 wouldn't accept that.

I think the best thing is to do is to close this issue, thanks for your help

**Comment by jeanfredrik** [comment]
Glad you figured it out! I don’t know if this was an issue specific to this addon or not, but if you think there should be something about it in the readme, let me know!

**Comment by christopher-francisco** [comment]
Hi, I tried everything in this thread and nothing worked. Can we re-open the issue?

I'm using `pnpm`.

My deps:
```
"@babel/cli": "7.12.13",
"@babel/plugin-proposal-object-rest-spread": "7.12.13",
"@babel/polyfill": "^7.12.0",
"@babel/core": "7.12.13",
"@babel/plugin-proposal-class-properties": "7.12.13",
"@babel/plugin-proposal-nullish-coalescing-operator": "7.12.13",
"@babel/plugin-proposal-optional-chaining": "7.12.13",
"@babel/preset-env": "7.12.13",
"@babel/preset-react": "7.12.13",
"@babel/register": "7.12.13",
"@storybook/addon-actions": "6.2.0-beta.9",
"@storybook/addon-contexts": "5.3.21",
"@storybook/addon-docs": "6.2.0-beta.9",
"@storybook/addon-essentials": "6.2.0-beta.9",
"@storybook/addon-knobs": "6.1.21",
"@storybook/addon-links": "6.2.0-beta.9",
"@storybook/addon-options": "5.3.21",
"@storybook/addon-storyshots": "6.1.21",
"@storybook/addons": "6.2.0-beta.9",
"@storybook/core": "6.1.21",
"@storybook/react": "6.2.0-beta.9",
"@whitespace/storybook-addon-html": "5.0.0",
```

**Comment by thorn0** [comment]
> Can we re-open the issue?
> "@storybook/core": "6.1.21",

This is an upstream issue in Storybook, not in this package. You should update Storybook.

**Comment by christopher-francisco** [comment]
Thank you

**Comment by maddesigns** [comment]
I’m running into this issue now that I upgraded to Storybook 6.3.7. I also tried `@babel/core: 7.12.13` and the resolutions block with yarn. Sadly it does not help. 

Any further ideas, what I can try to do?

**Comment by phil-isaacs** [comment]
I'm currently running into the same issue. I upgraded to Storybook 6.3.1 and the whole thing blows up.

I try upgrading babel/core to the latest which as of today is `"@babel/core": "^7.15.8",`

Is there any fix of this currently?

**Comment by tfrijsewijk** [comment]
> I’m running into this issue now that I upgraded to Storybook 6.3.7. I also tried `@babel/core: 7.12.13` and the resolutions block with yarn. Sadly it does not help.
> 
> Any further ideas, what I can try to do?

and

> I'm currently running into the same issue. I upgraded to Storybook 6.3.12 and the whole thing blows up.
> 
> I try upgrading babel/core to the latest which as of today is `"@babel/core": "^7.15.8",`
> 
> Is there any fix of this currently?

The problem lies with how Storybook manages its subdependencies. In an ideal world they would test functionality brought in by the packages they brought in and manage them for you, but that's a lot of work (no hate here, Storybook). So you need to help them/yourself.

As mentioned by [thorn0 it was fixed in Babel v7.12.13](https://github.com/whitespace-se/storybook-addon-html/issues/46#issuecomment-812122318) which meant changes in [babel-helper-create-class-features-plugin, babel-helper-replace-supers and babel-plugin-transform-classes](https://github.com/whitespace-se/storybook-addon-html/issues/46#issuecomment-813982371). So you need to update those packages in your local install: Your lock file forces an install to use the wrong/older version.

There are numerous ways to force an update of locked (sub)packages:
* [`yarn up @babel/*` ](https://yarnpkg.com/cli/up#examples) if you're using Yarn, I don't know how this works with npm
* [Force with resolutions](https://github.com/whitespace-se/storybook-addon-html/issues/46#issuecomment-815811891). Extra note: You only need to add these for one `yarn install`. When `yarn install` updates the lockfile you can remove them, the version range that Storybook uses for its dependencies doesn't need a forced resolution.
* Remove the entries from your lock file and delete `node_modules`.
* Nuclear option: delete your lock file (and `node_modules`). This means that ALL your dependencies are installed to the latest possible version -- use with caution!

**Comment by phil-isaacs** [comment]
Hey @tfrijsewijk Thanks for the feedback. 

I'm going to try and force my project to use the specific versions you sited above, but I'm a little unsure how to do this with my package.json file (my team is all using npm) and which packages I need to target here. 

Thanks again in advance,

Philip
